src/transformers/models/aria/modular_aria.py

```diff
@@ -0,0 +1,1357 @@
+                prompt_strings.append(sample)
+
+        else:
+            image_inputs = {}
+            prompt_strings = text
+
+        text_inputs = self.tokenizer(
+            prompt_strings,
+            return_tensors=output_kwargs["text_kwargs"]["return_tensors"],
+            padding=output_kwargs["text_kwargs"]["padding"],
+            truncation=output_kwargs["text_kwargs"]["truncation"],
+            max_length=output_kwargs["text_kwargs"]["max_length"],
+        )
+
+        return BatchFeature(data={**text_inputs, **image_inputs})
+
+    def save_pretrained(self, save_directory, **kwargs):
+        """
+        Save both the image processor and tokenizer.
+        """
+        merged_kwargs = self._merge_kwargs(
+            AriaProcessorKwargs,
+            {},
+            **kwargs,
+        )
+        if self.image_processor is not None:
+            self.image_processor.save_pretrained(
+                save_directory,
+                **merged_kwargs["images_kwargs"],
+            )
+        if self.tokenizer is not None:
+            self.tokenizer.save_pretrained(
+                save_directory,
+                **merged_kwargs["text_kwargs"],
+            )
+
+    @classmethod
+    def from_pretrained(
+        cls,
+        pretrained_model_name_or_path,
+        tokenizer_path=None,
+        image_processor_path=None,
+        **kwargs,
+    ):
+        """
+        Load both the image processor and tokenizer from a pretrained model path.
+        """
+        tokenizer_path = tokenizer_path if tokenizer_path is not None else pretrained_model_name_or_path
+        image_processor_path = (
+            image_processor_path if image_processor_path is not None else pretrained_model_name_or_path
+        )
+        image_processor = AriaImageProcessor.from_pretrained(
+            image_processor_path,
+            **kwargs,
+        )
+        if "use_fast" in kwargs:
+            logger.warning("use_fast is not supported for AriaProcessor. Ignoring...")
+            kwargs.pop("use_fast")
+        tokenizer = AutoTokenizer.from_pretrained(
+            tokenizer_path,
+            use_fast=False,
+            **kwargs,
+        )
+        chat_template = tokenizer.chat_template
+
+        return cls(
+            image_processor=image_processor,
+            tokenizer=tokenizer,
+            chat_template=chat_template,
+        )
+
+    # Copied from transformers.models.clip.processing_clip.CLIPProcessor.batch_decode with CLIP->Llama
+    def batch_decode(self, *args, **kwargs):
+        """
+        This method forwards all its arguments to LlamaTokenizerFast's [`~PreTrainedTokenizer.batch_decode`]. Please
+        refer to the docstring of this method for more information.
+        """
+        return self.tokenizer.batch_decode(*args, **kwargs)
+
+    # Copied from transformers.models.clip.processing_clip.CLIPProcessor.decode with CLIP->Llama
```

Comment:
Hmm, are the "copied from" statements transferred by modular if we inherit the class from ClipProcessor? Should be i think, but lmk if it doesn't and we'll improve modular in the next releases